### ISSUE TITLE
Error creating GL context; Received multiple errors. Errors: `[OsError("eglInitialize failed"), OsError("GL context creation failed")]`

### ISSUE THREAD

**Issue by rudolfschmidt** [alacritty/alacritty#4744]
After upgrading my system (guessing Nvidia packages) I cannot open the terminal anymore. It happens for a long time. Only an X restart may help, or even a full system restart.

### System

OS: `Linux home 5.10.10-arch1-1 #1 SMP PREEMPT Sat, 23 Jan 2021 23:59:48 +0000 x86_64 GNU/Linux`
Version: `alacritty 0.7.1 (a852eb85)`
Linux/BSD: `X1 i3`

### Logs

Font/Terminal size: 

```
$ alacritty -vv
Created log file at "/tmp/Alacritty-264803.log"
[2021-01-27 20:22:59.536655491] [INFO ] [alacritty] Welcome to Alacritty
[2021-01-27 20:22:59.536743547] [INFO ] [alacritty] Configuration files loaded from:
                                                      "/home/rs/.config/alacritty/alacritty.yml"
[2021-01-27 20:22:59.552463599] [DEBUG] [crossfont] Loaded Face Face { ft_face: Font Face: Regular, load_flags: TARGET_LIGHT, render_mode: "Lcd", lcd_filter: 1 }
[2021-01-27 20:22:59.553347681] [DEBUG] [alacritty] Estimated DPR: 1.1666666666666667
[2021-01-27 20:22:59.553379972] [DEBUG] [alacritty] Estimated window size: None
[2021-01-27 20:22:59.553389479] [DEBUG] [alacritty] Estimated cell size: 10 x 22
[2021-01-27 20:22:59.629051014] [ERROR] [alacritty] Alacritty encountered an unrecoverable error:

                                                    	Error creating GL context; Received multiple errors. Errors: `[OsError("eglInitialize failed"), OsError("GL context creation failed")]`
```

Keyboard and bindings: 
```
$ alacritty --print-events
Created log file at "/tmp/Alacritty-264900.log"
[2021-01-27 20:23:20.907745718] [INFO ] [alacritty] Welcome to Alacritty
[2021-01-27 20:23:20.907801552] [INFO ] [alacritty] Configuration files loaded from:
                                                      "/home/rs/.config/alacritty/alacritty.yml"
[2021-01-27 20:23:20.975406620] [ERROR] [alacritty] Alacritty encountered an unrecoverable error:

                                                    	Error creating GL context; Received multiple errors. Errors: `[OsError("eglInitialize failed"), OsError("GL context creation failed")]`

```

**Comment by chrisduerr** [comment]
Did you restart your system?

**Comment by rudolfschmidt** [comment]
> Did you restart your system?

not when I posted the post, after a restart the error was away. Do you know why this happens? It's a little bit unfortunate to restart the entire system just to use the terminal app again.

**Comment by chrisduerr** [comment]
Some drivers require a restart to work, that's normal and not something up to Alacritty.

**Comment by robclancy** [comment]
Having to restart your system because of a terminal on linux is a joke. No other program has this problem.

**Comment by GoodGuyPat** [comment]
> Having to restart your system because of a terminal on linux is a joke. No other program has this problem.

If you choose a GPU rendered terminal emulator, you are subject to the functionality of your GPU, which relies on drivers. Alacritty is offered to you for free and it is a very solid piece of software. The level of entitlement of users who feel like they have a right to complain as if they were paying customers, from a position of (understandable) ignorance, only hurts the open source community and demotivates those who volunteer their time and work so that we may have this software.

Thank you, devs!

**Comment by dym-sh** [comment]
approx. things needed for proper OpenGL context (same happened with zoom.deb)
```sh
sudo apt-get install -y \
  ibus ibus-data ibus-gtk ibus-gtk3 \
  libegl1-mesa libgl1-mesa-glx \
  libxcb-xtest0 python3-ibus-1.0
```

**Comment by kchibisov** [comment]
You don't need a reboot for mesa for sure, you might need login/logout here though. And not sure how Xorg handles that, but on Wayland I update drivers without any reboot (only kernel update required it for sure).

**Comment by chrisduerr** [comment]
@kchibisov It depends on the driver. There are some shitty graphics drivers that require a reboot.

**Comment by kchibisov** [comment]
I'm not sure how that could be, since the driver is just a userspace lib? Though, maybe there's such a thing for some of them.

**Comment by doc1623** [comment]
I can't speak for the intent, of @rudolfschmidt comment, but I somewhat agree. That's not a dig at alacrity, though. I've enjoyed using it. It's awesome (Thank YOU). I'm not an expert, but I do seem to need to reboot more with systemd, than I did without, and on non-system systemd distros. I'm using the ubuntu that came on mine (from dell) and it started replacing my apt install programs with their snap :( and I start seeing memory leaks that I didn't have before (firefox). Before I break down and restart (as I have too much stuff open that I like switching to/from. I'll see if I can restart the driver. Don't know if that'll work but I'll try.
